user-service/models/repository.js

```diff
@@ -6,11 +6,11 @@ export async function _createUser(param) {
 }
 
 export async function _deleteUser(param) {
-    return UserModel.findOneAndDelete({ email: param });
+    return UserModel.findOneAndUpdate({ email: param }, { isDeleted: true }, { new: true }); // soft delete
 }
 
 export async function _findUser(param) {
-    return UserModel.findOne({ email: param });
+    return UserModel.findOne({ email: param, isDeleted: false });
 }
 
 export async function _updateUser({ email, prop }) {
```

Comment:
Modifying this update method to check if isDeleted is false will resolve issue of not updating correct user